src/XMakeTasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -592,4 +599,170 @@ private void FreeAsmMeta(IntPtr asmMetaPtr)
             }
         }
     }
+
+    /// <summary>
+    /// Managed implementation of a reader for getting the runtime version of an assembly
+    /// </summary>
+    static class ManagedRuntimeVersionReader
+    {
+        class HeaderInfo
+        {
+            public uint VirtualAddress;
+            public uint Size;
+            public uint FileOffset;
+        }
+
+        /// <summary>
+        /// Given a path get the CLR runtime version of the file
+        /// </summary>
+        /// <param name="path">path to the file</param>
+        /// <returns>The CLR runtime version or empty if the path does not exist or the file is not an assembly.</returns>
+        public static string GetRuntimeVersion(string path)
+        {
+            using (var sr = new BinaryReader(File.OpenRead(path)))
+            {
+                if (!File.Exists(path))
+                    return string.Empty;
+                
+                try
+                {
+                    const uint PEHeaderPointerOffset = 0x3c;
+                    const uint PEHeaderSize = 20;
+                    const uint OptionalPEHeaderSize = 224;
+                    const uint OptionalPEPlusHeaderSize = 240;
+                    const uint SectionHeaderSize = 40;
+
+                    // There must be room for at least the PE header offset and size, and sections
+                    if (sr.BaseStream.Length < PEHeaderPointerOffset + 4 + PEHeaderSize + OptionalPEHeaderSize + SectionHeaderSize)
+                        return string.Empty;
+
+                    // PE header
+                    sr.BaseStream.Position = PEHeaderPointerOffset;
+                    var peHeaderOffset = sr.ReadUInt32();
```

Comment:
Since you reuse this variable for various jumps through the header, perhaps it would be more clear if the name reflects that it is the current position?